iam-policy-autopilot-lints/src/node_kind_literal.rs

```diff
@@ -38,10 +38,18 @@ dylint_linting::declare_late_lint! {
     "use of string literals in comparisons with .kind() method calls"
 }
 
-/// Check if an expression is a call to the `.kind()` method
-fn is_kind_method_call(expr: &Expr<'_>) -> bool {
-    if let ExprKind::MethodCall(path_segment, _, _, _) = &expr.kind {
-        path_segment.ident.name.as_str() == "kind"
+/// Check if an expression is a call to the `.kind()` method on a tree-sitter Node
+fn is_kind_method_call<'tcx>(cx: &rustc_lint::LateContext<'tcx>, expr: &'tcx Expr<'_>) -> bool {
+    if let ExprKind::MethodCall(path_segment, receiver, _, _) = &expr.kind {
+        if path_segment.ident.name.as_str() != "kind" {
+            return false;
+        }
+
+        // Check if the receiver type contains "Node" in its path
+        // This catches tree_sitter::Node and similar types
```

Comment:
Given that we don't have unit tests using tree_sitter, I trust you checked that this is the right thing to check for?